pyomo/repn/plugins/gams_writer.py

```diff
@@ -160,7 +160,7 @@ def ctype(self, comp):
         if isinstance(comp, ICategorizedObject):
             return comp.ctype
         else:
-            return comp.type()
+            return comp.ctype
```

Comment:
If the name of the `ctype` property is the same as kernel components now, code like this (found at least in the GAMS and Baron writers) can probably be removed. It looks like this line is part of a helper function meant specifically for this workaround.